docassemble/AssemblyLine/custom_jinja_filters.py

```diff
@@ -200,11 +203,140 @@ def catchall_subquestion(value: Any, subquestion: str) -> DACatchAll:
     return value
 
 
+def _undefined_label(ud: Undefined, placeholder: Optional[str] = None) -> str:
+    """
+    Extract a friendly name from a Jinja2 Undefined.
+    Prefer the last missing attribute/key; fall back to the variable name;
+    then try parsing the hint; finally a generic label.
+
+    Args:
+        ud (Undefined): The Jinja2 Undefined object.
+        placeholder (str, optional): A generic label to use if no other name can be found.
```

Comment:
Since `placeholder` isn't ever passed in, should it be removed, and the fallback just be "undefined"?